lambda/matching.ml

```diff
@@ -3106,22 +3105,26 @@ let combine_constant value_kind loc arg cst partial ctx def
 
 let split_cases tag_lambda_list =
   let rec split_rec = function
-    | [] -> ([], [])
+    | [] -> ([], [], [])
     | ({cstr_tag; cstr_repr; cstr_constant}, act) :: rem -> (
-        let consts, nonconsts = split_rec rem in
+        let consts, nonconsts, nulls = split_rec rem in
         match cstr_tag, cstr_repr with
-        | Ordinary _, Variant_unboxed -> (consts, (0, act) :: nonconsts)
+        | Ordinary _, (Variant_unboxed | Variant_with_null) ->
+          (consts, (0, act) :: nonconsts, nulls)
```

Comment:
This is a bit misleading, Because the value is not (necessarily) a block with tag 0, but the problem is exactly the same with `This` as with unboxed variants, so I'm fine with the code.

Reply:
Just to be sure, can this lead to any incorrect behavior?

Reply:
No, for the same reason as unboxed variants: there will be no `const` case and no other `non_const` case, so we will never generate any branching instruction depending on this tag.